src/Npgsql/NpgsqlException.cs

```diff
@@ -45,6 +45,15 @@ public NpgsqlException(string message, Exception innerException)
         public NpgsqlException(string message)
             : base(message) { }
 
+        public virtual bool IsTransient
+        {
+            get
+            {
+                return this.InnerException != null && (this.InnerException.GetType() == typeof(System.IO.IOException));
```

Comment:
I'm not sure how the ?? operator would apply here.  I'm not trying to return the value of the inner exception.  As for the other points, I will work them in.